File: problem023.py
import math

def factorSum(n):
	s = 1
	i = 2
	l = int(math.sqrt(n))
	while i < l:
		if n % i == 0:
			s += i
			s += n / i
		i += 1
	if n % l == 0:
		s += l
	return s



abundant = []

i = 1
while i <= 28123:
	if factorSum(i) > i:
		abundant.append(i)
	i += 1
print(len(abundant))
asAbSum = set()
for i in abundant:
	for j in abundant:
		if i+j <= 28123:
			asAbSum.add(i+j)
		else:
			break
s = 0
i = 1
while i <= 28123:
	if i not in asAbSum:
		s += i
	i += 1
print(s)

File: problem023.rb
def factorSum(n)
	sum = 1
	l = Integer(Math.sqrt(n))
	(2...l).each do |i|
		if n % i == 0
			sum += i
			sum += n / i
		end
		i += 1
	end
	sum += l if n % l == 0
	sum
end

abundant = []
(1..28123).each do |i|
	if factorSum(i) > i
		abundant.push(i)
	end
end


puts(abundant.length)

asAbSum = Hash.new true

abundant.each do |i|
	for j in abundant do
		if i+j <= 28123
			asAbSum[i+j]= false
		else
			break
		end
	end
end
sum = 0
for i in 1..28123
	if asAbSum[i]
		sum += i
	end
end
puts(sum)